src/test/java/streams/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,127 @@
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toList()
+        List<Person> personsEverWorkedInEpam = employees.stream()
+                                                        .filter(employee -> employee.getJobHistory()
+                                                                                    .stream()
+                                                                                    .anyMatch(entry -> entry.getEmployer()
+                                                                                                            .equalsIgnoreCase("EPAM")))
+                                                        .map(Employee::getPerson)
+                                                        .collect(Collectors.toList());
+
+        assertThat(personsEverWorkedInEpam, contains(
+                employees.get(0).getPerson(),
+                employees.get(1).getPerson(),
+                employees.get(4).getPerson(),
+                employees.get(5).getPerson())
+        );
+    }
+
+    @Test
+    void findPersonsBeganCareerInEpam() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toList()
+        List<Person> startedFromEpam = employees.stream()
+                                                .filter(employee -> {
+                                                    Optional<JobHistoryEntry> firstJob = employee.getJobHistory().stream().findFirst();
+                                                    return firstJob.isPresent() && firstJob.get().getEmployer().equalsIgnoreCase("EPAM");
+                                                })
+                                                .map(Employee::getPerson)
+                                                .collect(Collectors.toList());
+        assertThat(startedFromEpam, contains(
+                employees.get(0).getPerson(),
+                employees.get(1).getPerson(),
+                employees.get(4).getPerson()
+        ));
+    }
+
+    @Test
+    void findAllCompanies() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toSet()
+        Set<String> companies = employees.stream()
+                                         .flatMap(employee -> employee.getJobHistory().stream())
+                                         .map(JobHistoryEntry::getEmployer)
+                                         .distinct()
+                                         .collect(Collectors.toSet());
+
+        assertThat(companies, containsInAnyOrder("EPAM", "google", "yandex", "mail.ru", "T-Systems"));
+    }
+
+    @Test
+    void findMinimalAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация
+        Integer minimalAge = employees.stream()
+                                      .min(Comparator.comparingInt(e -> e.getPerson().getAge()))
```

Comment:
It would be more readable, if you wrote all `.map(...)` functions at first and then compare the values.